temporalproto/public_methods.go

```diff
@@ -20,27 +20,32 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-// This code taken from the bottom of
-// https://github.com/grpc-ecosystem/grpc-gateway/blob/v1.16.0/runtime/query.go.
-
-package temporalgateway
+package temporalproto
 
 import (
 	"reflect"
-
-	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"regexp"
 )
 
-var (
-	convFromType = map[reflect.Kind]reflect.Value{
-		reflect.String:  reflect.ValueOf(runtime.String),
-		reflect.Bool:    reflect.ValueOf(runtime.Bool),
-		reflect.Float64: reflect.ValueOf(runtime.Float64),
-		reflect.Float32: reflect.ValueOf(runtime.Float32),
-		reflect.Int64:   reflect.ValueOf(runtime.Int64),
-		reflect.Int32:   reflect.ValueOf(runtime.Int32),
-		reflect.Uint64:  reflect.ValueOf(runtime.Uint64),
-		reflect.Uint32:  reflect.ValueOf(runtime.Uint32),
-		reflect.Slice:   reflect.ValueOf(runtime.Bytes),
+var publicMethodRgx = regexp.MustCompile("^[A-Z]")
+
+// PublicMethods calls the provided callback on each method declared as public on the
+// specified object.
+// This prevents the `mustEmbedUnimplementedFooBarBaz` method required by the GRPC v2
+// gateway from polluting our tests.
+func PublicMethods(obj any, cb func(reflect.Method)) {
```

Comment:
How is this used? Must this non-Temporal-non-proto utility be exposed as public API from the Temporal proto library?

Reply:
These should have stayed in the server, whoops.

Reply:
I'd like to keep DeepEqual here at least; it's complex enough that I really don't want us maintaining multiple copies of it (and I'm using it in the CLI's testing)

Reply:
Since it is very Temporal proto specific, being in `temporalproto` makes sense for that one.